docs/sources/setup-grafana/installation/docker/index.md

```diff
@@ -8,253 +8,318 @@ weight: 200
+```
 
-The default images are based on the popular [Alpine Linux project](http://alpinelinux.org), available in [the Alpine official image](https://hub.docker.com/_/alpine). Alpine Linux is much smaller than most distribution base images, and thus leads to slimmer and more secure images.
+where:
 
-The Alpine variant is highly recommended when security and final image size being as small as possible is desired. The main caveat to note is that it uses [musl libc](http://www.musl-libc.org) instead of [glibc and friends](http://www.etalabs.net/compare_libcs.html), so certain software might run into issues depending on the depth of their libc requirements. However, most software don't have an issue with this, so this variant is usually a very safe choice.
+run = run directly from the command line\
+d = run in the background\
+p = assign the port number, which in this case is `3000`\
+name = assign a logical name to the container, for example, `grafana`\
+grafana/grafana-enterprise = the image to run in the container
 
-> **Note:** Grafana docker images were based on [Ubuntu](https://ubuntu.com/) prior to version 6.4.0.
+### Stop the Grafana container
 
-## Ubuntu image
+To stop the Grafana container, run the following command:
 
-**Grafana Enterprise edition**: `grafana/grafana-enterprise:<version>-ubuntu`
+```bash
+# docker ps command tells about the running process in the docker
+docker ps
 
-**Grafana Open Source edition**: `grafana/grafana-oss:<version>-ubuntu`
+# This will display a list of containers, like:
+CONTAINER ID   IMAGE  COMMAND   CREATED  STATUS   PORTS    NAMES
+cd48d3994968   grafana/grafana-enterprise   "/run.sh"   8 seconds ago   Up 7 seconds   0.0.0.0:3000->3000/tcp   grafana
 
-These images are based on [Ubuntu](https://ubuntu.com/), available in [the Ubuntu official image](https://hub.docker.com/_/ubuntu). It is an alternative image for those who prefer an [Ubuntu](https://ubuntu.com/) based image and/or are dependent on certain tooling not available for Alpine.
+# To stop the grafana container run the command
+# docker stop CONTAINER-ID or use 
+# docker stop NAME (which is grafana as we defined previously)
+docker stop grafana
+```
 
-## Run Grafana
+### Save your Grafana data
 
-You can run the latest Grafana version, run a specific version, or run an unstable version based on the main branch of the [grafana/grafana GitHub repository](https://github.com/grafana/grafana).
+When you use Docker containers, the data inside them is temporary by default. This means that if you don't specify where to store the information, all the Grafana data will be lost when you stop the Docker container. To avoid losing your data, you can set up [persistent storage](https://docs.docker.com/storage/volumes/) or [bind mounts](https://docs.docker.com/storage/bind-mounts/) for your container.
 
-### Run the latest stable version of Grafana
+#### Use persistent storage (recommended)
 
```

Comment:
@usmangt - When would I do this and why is this recommended?

Reply:
It is recommended because running without it, users will lose all the data once the container is shut down. Users should use this method if thy are deploying in a production environment or where the data is critical.

Reply:
Let's add that information in the intro to this section, and call out why persistent storage is recommended over bind mounts.

Reply:
Done, added some information. Please check

Reply:
Ok, I've reviewed what you've added. When I read the second sentence, "Users should use this method if..." it makes me believe that bind mounts cannot be used in a production environment. Is that true?

Reply:
Thanks for the great documentation @usmangt @chri2547 
I was wondering if there is a an overlap between Docker Volumes / External Database / High Availability setup / Backup-Restore:

- If I as an admin do not use Docker Volumes but an external database, which data is lost after a container shut down?
- Do the files in Docker Volumes need to be replicated across all Grafana instances in an HA setup?
- Do the files in Docker Volumes need to be part of a backup/restore procedure if an external database is used?

Reply:
Hi @justjulian ,

Thanks for the feedback :slightly_smiling_face: 

I will try to answer your questions to the best knowledge.

> - If I as an admin do not use Docker Volumes but an external database, which data is lost after a container shut down?

So using persistent storage will not allow to data loss. Now, running docker as a non-root user, you might need to check it at this [link](https://docs.docker.com/engine/install/linux-postinstall/#manage-docker-as-a-non-root-user) as how to set it up correctly. Once done, you have to pass the current non-root user-ID in the `docker` or `docker compose`. Here is an example which I also answered in the [community post](https://community.grafana.com/t/new-docker-install-with-persistent-storage-permission-problem/10896/18?u=usman.ahmad)


> - Do the files in Docker Volumes need to be replicated across all Grafana instances in an HA setup?

This is a bit of an expert-level topic. What I know is that for reaching HA for Grafana you also might need to consider using Docker Swarm which can give you such level of DB replication (replicaset) if a node goes down. Found this reference [link](https://dockerlabs.collabnix.com/intermediate/Implementing_High_Availability_with_Docker_Swarm.html) as it might help to understand more.


> - Do the files in Docker Volumes need to be part of a backup/restore procedure if an external database is used?

This is also an expert-level question. Unfortunately, I do not have the answer to this question. What I can tell from experience is that this needs to be thoroughly tested because if you are e.g. using a 3rd party Database which have a different storage Engine, it may work and behave differently on different public cloud networks e..g AWS, GCP; Azure, Alibaba, etc. I would recommend using the [Docker Community forum](https://forums.docker.com/) for this one.

Reply:
Thanks @usmangt , yes, those insights help a lot.
I think all three questions can probably be summarized as: Is there any data loss if not using _Docker Volumes / persistent storage / bind mounts_ but an external database instead? What are your thoughts on that?

Reply:
Thanks @DanCech that clarifies it perfectly :)